From f97b71158fe10fb585267857c16ee9deb8c6ccc4 Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Mon, 11 Jan 2021 09:23:09 -0600
Subject: [PATCH] Fixed test on x86 RH8

---
 official/recommendation/data_preprocessing.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/official/recommendation/data_preprocessing.py b/official/recommendation/data_preprocessing.py
index 4fc510b..59222ef 100644
--- a/official/recommendation/data_preprocessing.py
+++ b/official/recommendation/data_preprocessing.py
@@ -196,7 +196,7 @@ def _filter_index_sort(raw_rating_path: Text,
 
     logging.info("Writing raw data cache.")
     with tf.io.gfile.GFile(cache_path, "wb") as f:
-      pickle.dump(data, f, protocol=pickle.HIGHEST_PROTOCOL)
+      pickle.dump(data, f, protocol=pickle.DEFAULT_PROTOCOL)
 
   # TODO(robieta): MLPerf cache clear.
   return data, valid_cache
-- 
1.8.3.1

