From 967299bcddcaa2eeaccd842061a47c41475c1f38 Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Tue, 4 Apr 2023 11:50:25 +0000
Subject: [PATCH] hwloc updated for s390x

---
 third_party/hwloc/hwloc.BUILD | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/third_party/hwloc/hwloc.BUILD b/third_party/hwloc/hwloc.BUILD
index 6d316eb1f83..9ab86258725 100644
--- a/third_party/hwloc/hwloc.BUILD
+++ b/third_party/hwloc/hwloc.BUILD
@@ -273,6 +273,10 @@ cc_library(
             "hwloc/topology-linux.c",
             "include/hwloc/linux.h",
         ],
+        "@org_tensorflow//tensorflow:linux_s390x": [
+            "hwloc/topology-linux.c",
+            "include/hwloc/linux.h",
+        ],
         "@org_tensorflow//tensorflow/tsl:freebsd": [
             "hwloc/topology-freebsd.c",
             "hwloc/topology-x86.c",
-- 
2.34.1

